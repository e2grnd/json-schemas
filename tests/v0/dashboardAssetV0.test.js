const fs = require('fs');
const path = require('path');
const Ajv = require('ajv/dist/2020').default;
const ajv = new Ajv({ strict: false, allErrors: true});

const dashboardAssetSchema = require('../../v0/dashboard.asset.schema.json');
const validate_dashboard_asset = ajv.compile(dashboardAssetSchema);


describe('Asset Dashboard data validation', () => {
  it('validates passing data against the schema', () => {
    const passingData = require('../../data/v0/passing_dashboard_asset_v0.json');
    const isValid = validate_dashboard_asset(passingData);
    expect(validate_dashboard_asset.errors).toBeNull();
    expect(isValid).toBeTruthy(); // Assert that the data passes validation
  });

  it('validates failing data against the schema', () => {
    const failingData = require('../../data/v0/failing_dashboard_asset_v0.json');
    const isValid = validate_dashboard_asset(failingData);
    expect(validate_dashboard_asset.errors).toMatchSnapshot(); // Assert that the data fails validation
  });
})
