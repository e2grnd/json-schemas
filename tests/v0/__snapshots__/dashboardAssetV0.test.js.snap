// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Asset Dashboard data validation validates failing data against the schema 1`] = `
[
  {
    "instancePath": "/layout/0",
    "keyword": "required",
    "message": "must have required property 'x'",
    "params": {
      "missingProperty": "x",
    },
    "schemaPath": "#/properties/layout/items/required",
  },
  {
    "instancePath": "/layout/0",
    "keyword": "required",
    "message": "must have required property 'y'",
    "params": {
      "missingProperty": "y",
    },
    "schemaPath": "#/properties/layout/items/required",
  },
  {
    "instancePath": "/layout/0",
    "keyword": "required",
    "message": "must have required property 'w'",
    "params": {
      "missingProperty": "w",
    },
    "schemaPath": "#/properties/layout/items/required",
  },
]
`;
